File: notification_hub_rest/Sample/NotificationHubSample/NotificationHubSample/Models/DeviceInstallation.swift
//
//  DeviceInstallation.swift
//  NotificationHubSample
//
//  Created by Mike Parker on 04/05/2018.
//  Copyright © 2018 mobcat. All rights reserved.
//

import Foundation

struct DeviceInstallation : Codable {
    let installationId : String
    let pushChannel : String
    let platform : String = "apns"
    var tags : [String]
    var templates : Dictionary<String, PushTemplate>
    
    init(withInstallationId installationId : String, andPushChannel pushChannel : String) {
        self.installationId = installationId
        self.pushChannel = pushChannel
        self.tags = [String]()
        self.templates = Dictionary<String, PushTemplate>()
    }
}

File: notification_hub_backend_service/src/azure/PushDemoApi/PushDemoApi/Models/DeviceInstallation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;

namespace PushDemoApi.Models
{
    public class DeviceInstallation
    {
        [Required]
        public string InstallationId { get; set; }

        [Required]
        public string Platform { get; set; }

        [Required]
        public string PushChannel { get; set; }

        public IList<string> Tags { get; set; } = Array.Empty<string>();

        public Dictionary<string, PushTemplate> Templates { get; set; } = new Dictionary<string, PushTemplate>();
    }
}